include/swift/AST/Mangle.h

```diff
@@ -154,6 +154,8 @@ class Mangler {
   void mangleProtocolList(ArrayRef<Type> protocols);
   void mangleIdentifier(Identifier ident,
                         OperatorFixity fixity = OperatorFixity::NotOperator);
+  void mangleIdentifier(DeclBaseName ident,
```

Comment:
I would imagine that we will at some point change the mangling name of subscripts to something that doesn't contain the string "subscript" (I'd use a new special character in the mangling scheme). Then the special names would be handled on the call side of this function and we could remove it again.